language/sygus.tex

```diff
@@ -1816,14 +1817,16 @@ \subsubsection{Weights}
 
 Let $y \mapsto_{W,k} t_0$ denote a production rule
 whose weight with respect to $W$ is $k$.
-When the body of function-to-synthesize $f$ is interpreted as term $r$,
-the symbol \weightsymnf{$w$}{$f$} can be interpreted as the value 
-$k_1 + \ldots + k_n$ if it is possible to construct a sequence of terms
+We write $\wgenerates{W}{k}{r} \in G$ if it is possible to construct a sequence of terms
```

Comment:
Minor (and subjective), but I think the rendering for this macro is a bit awkward to read. What are your thoughts on:

```tex
%definition
\global\long\def\wgenerates#1#2#3#4{#1 \mapsto_{#2,#3}^{\ast} #4}

%usage
\wgenerates{G}{W}{k}{r}
```

Basically we would use a style that's very similar to nonterminals, and it looks intuitive when reading.

Reply:
Yes, I agree with this change.